setup.py

```diff
@@ -58,7 +62,7 @@ def _maintain_symlinks(symlink_type, base_path):
         # Try the cache first because going from git checkout to sdist is the
         # only time we know that we're going to cache correctly
         with open(SYMLINK_CACHE, 'r') as f:
-            symlink_data = json.loads(f.read())
+            symlink_data = json.load(f)
```

Comment:
Just a note... We probably won't run into problems here as don't currently have symlinks with non-ascii filenames but this would be an issue if we did.  I can't remember all the permutations between py2.6 and py3 but I think we'd either have to play around with the ensure_ascii and encoding arguments of json.load or do our own encoding and decoding and probably go back to f.write() and f.read()  instead of using json.dump() and json.load().

I'm mentioning it since there's definitely places in the ansible code itself, where that is an issue so it's not a safe refactor in general.